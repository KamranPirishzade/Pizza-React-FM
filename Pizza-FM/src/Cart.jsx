import useCurrency from "./useCurrency";

const Cart = ({ cart, checkout }) => {
  let total = cart.reduce((initial, item) => {
    return initial + item.pizza.sizes[item.size];
  }, 0);

  return (
    <div className="cart">
      <h2>Cart</h2>
      <ul>
        {cart.map((item, index) => (
          <li key={index}>
            <span className="size">{item.size}</span>-
            <span className="type">{item.pizza.name}</span>-
            <span className="price">{item.price}</span>
          </li>
        ))}
      </ul>
      <p>Total:{useCurrency(total)}</p>
      <button onClick={checkout}>Checkout</button>
    </div>
  );
};

export default Cart;
