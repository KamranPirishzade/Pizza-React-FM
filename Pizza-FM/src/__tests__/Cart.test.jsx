import { test, expect } from "vitest";
import { render } from "@testing-library/react";
import Cart from "../Cart";

test("Check cart snapshot when nothing given to component", () => {
  const { asFragment } = render(<Cart cart={[]} />);
  expect(asFragment()).toMatchSnapshot();
});


//Can be used for api correct api integration with back 
test("test request data type", () => {
  expect({
    id: "calabrese",
    name: "The Calabrese Pizza",
    category: "Supreme",
    description:
      "Salami, Pancetta, Tomatoes, Red Onions, Friggitello Peppers, Garlic",
    image: "/public/pizzas/calabrese.webp",
    sizes: { S: 12.25, M: 16.25, L: 20.25 },
  }).toMatchSnapshot();
});
