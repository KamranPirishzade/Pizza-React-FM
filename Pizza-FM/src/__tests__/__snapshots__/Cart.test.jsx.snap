// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Check cart snapshot when nothing given to component 1`] = `
<DocumentFragment>
  <div
    class="cart"
  >
    <h2>
      Cart
    </h2>
    <ul />
    <p>
      Total:$0.00
    </p>
    <button>
      Checkout
    </button>
  </div>
</DocumentFragment>
`;

exports[`test request data type 1`] = `
{
  "category": "Supreme",
  "description": "Salami, Pancetta, Tomatoes, Red Onions, Friggitello Peppers, Garlic",
  "id": "calabrese",
  "image": "/public/pizzas/calabrese.webp",
  "name": "The Calabrese Pizza",
  "sizes": {
    "L": 20.25,
    "M": 16.25,
    "S": 12.25,
  },
}
`;
